starlite/config.py

```diff
@@ -33,9 +34,47 @@ class CORSConfig(BaseModel):
     max_age: int = 600
 
 
-class GZIPConfig(BaseModel):
+class CompressionBackend(str, Enum):
+    GZIP = "gzip"
+    BROTLI = "brotli"
+
+
+class BrotliMode(str, Enum):
+    GENERIC = "generic"
+    TEXT = "text"
+    FONT = "font"
+
+
+class CompressionConfig(BaseModel):
+    backend: Optional[CompressionBackend] = None
     minimum_size: int = 500
-    compresslevel: int = 9
+    gzip_compress_level: int = 9
+    brotli_quality: int = 5
+    brotli_mode: BrotliMode = BrotliMode.TEXT
+    brotli_lgwin: int = 22
+    brotli_lgblock: int = 0
+    brotli_gzip_fallback: bool = True
+    CompressionMiddleware: PyObject = None
+
+    def __init__(__pydantic_self__, **data: Any) -> None:
+        backend = data.get("backend", None)
```

Comment:
No need for the None default, because it is the default.

Reply:
Removed the `None` after setting backend to required.